test/e2e/tests/add-hide-token.spec.js

```diff
@@ -123,3 +123,119 @@ describe('Add existing token using search', function () {
+  };
+
+  it('opens a notification that adds a token when wallet_watchAsset is executed, then approves', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.openNewPage('http://127.0.0.1:8080/');
+
+        await driver.executeScript(`
+          window.ethereum.request({
+            method: 'wallet_watchAsset',
+            params: {
+              type: 'ERC20',
+              options: {
+                address: '0x86002be4cdd922de1ccb831582bf99284b99ac12',
+                symbol: 'TST',
+                decimals: 4
+              },
+            }
+          })
+        `);
+
+        const windowHandles = await driver.waitUntilXWindowHandles(3);
+
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+
+        await driver.clickElement({
+          tag: 'button',
+          text: 'Add token',
+        });
+
+        await driver.switchToWindowWithTitle('MetaMask', windowHandles);
+
+        await driver.waitForSelector({
+          css: '.asset-list-item__token-value',
+          text: '0',
+        });
+
+        await driver.waitForSelector({
+          css: '.asset-list-item__token-symbol',
+          text: 'TST',
+        });
+      },
+    );
+  });
+
+  it('opens a notification that adds a token when wallet_watchAsset is executed, then rejects', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.openNewPage('http://127.0.0.1:8080/');
+
+        await driver.executeScript(`
+          window.ethereum.request({
+            method: 'wallet_watchAsset',
+            params: {
+              type: 'ERC20',
+              options: {
+                address: '0x86002be4cdd922de1ccb831582bf99284b99ac12',
+                symbol: 'TST',
+                decimals: 4
+              },
+            }
+          })
+        `);
+
+        const windowHandles = await driver.waitUntilXWindowHandles(3);
+
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+
+        await driver.clickElement({
+          tag: 'button',
+          text: 'Cancel',
+        });
```

Comment:
Should we be asserting something at this point? Such as that there is one less window, or that our original request promise was rejected?

Reply:
I couldn't see what we could be asserting here. By pressing reject it just won't add the currency, but there's no e2e helper method that I could find to check that an element has not been added. The parent element is also not particularly helpful in this case (no clear list of elements underneath to check assert the length).

When I checked other e2e tests such as the add custom network, it just presses the cancel button and doesn't assert anything afterwards.

Reply:
Added additional validation.